### PR TITLE
Fixed #29260 -- Skipped an UPDATE when adding a model instance with primary key that has a default.

### PR DESCRIPTION
**Pull request by hramezani**
[ticket 29260](https://code.djangoproject.com/ticket/29260)

### CODE REVIEW

File: tests/basic/models.py [hunk 1]
```diff
@@ -40,3 +42,12 @@ def __str__(self):
...
         return SelfRef.objects.get(selfref=self).pk
+
+
+class PrimaryKeyWithDefaultValue(models.Model):
+    id = models.UUIDField(
+        primary_key=True,
+        default=uuid.uuid4,
+        editable=False
+    )
+    title = models.TextField()
```
Comment (charettes) [inline tests/basic/models.py]: What's the purpose of this field?

File: tests/basic/tests.py [hunk 2]
```diff
@@ -130,6 +133,12 @@ def test_querysets_checking_for_membership(self):
         # ... but there will often be more efficient ways if that is all you need:
         self.assertTrue(Article.objects.filter(id=a.id).exists())
 
+    def test_save_primary_with_default_query_count(self):
+        obj = PrimaryKeyWithDefaultValue()
+        obj.title = 'abc'
```
Comment (charettes) [inline tests/basic/tests.py]: I think you can drop `title` entirely.

File: django/db/models/base.py [hunk 3]
```diff
@@ -842,6 +843,13 @@ def _save_table(self, raw=False, cls=None, force_insert=False,
         if not pk_set and (force_update or update_fields):
             raise ValueError("Cannot force an update in save() with no primary key.")
         updated = False
+
+        if (
+            not force_insert and self._state.adding and
+            self._meta.pk.default and self._meta.pk.default != NOT_PROVIDED
```
Comment (charettes) [inline django/db/models/base.py]:
```suggestion
            self._meta.pk.default and self._meta.pk.default is not NOT_PROVIDED
```

**Comment by hramezani** [comment]
@charettes Thanks. requested changes are done.

**Comment by felixxm** [comment]
@hramezani Thanks :+1:

**Pull request merged**